middleware/src/read_response_body.rs

```diff
@@ -0,0 +1,111 @@
+use actix_service::{Service, Transform};
+use actix_web::body::{BodySize, MessageBody, ResponseBody};
+use std::marker::PhantomData;
+use actix_web::{dev::ServiceRequest, dev::ServiceResponse, Error};
+use bytes::{Bytes, BytesMut};
+use futures::Async;
+use futures::future::{ok, FutureResult};
+use futures::{Future, Poll};
+
+pub struct Logging;
+
+impl<S: 'static, B> Transform<S> for Logging
+where
+    S: Service<Request = ServiceRequest, Response = ServiceResponse<B>, Error = Error>,
+    B: MessageBody + 'static,
+{
+    type Request = ServiceRequest;
+    type Response = ServiceResponse<BodyLogger<B>>;
+    type Error = Error;
+    type InitError = ();
+    type Transform = LoggingMiddleware<S>;
+    type Future = FutureResult<Self::Transform, Self::InitError>;
+
+    fn new_transform(&self, service: S) -> Self::Future {
+        ok(LoggingMiddleware {
+            service,
+        })
+    }
+}
+
+pub struct LoggingMiddleware<S> {
+    service: S,
+}
+
+impl<S, B> Service for LoggingMiddleware<S>
+where
+    S: Service<Request = ServiceRequest, Response = ServiceResponse<B>, Error = Error>,
+    B: MessageBody,
+{
+    type Request = ServiceRequest;
+    type Response = ServiceResponse<BodyLogger<B>>;
+    type Error = Error;
+    type Future = WrapperStream<S, B>;
+
+    fn poll_ready(&mut self) -> Poll<(), Self::Error> {
+        self.service.poll_ready()
+    }
+
+    fn call(&mut self, req: ServiceRequest) -> Self::Future {
+        WrapperStream {
```

Comment:
you can use box instead

Reply:
Sorry, use `Box` instead of what exactly? I surely can't replace the whole `WrapperStream` with a `Box`?

Reply:
I merged it but I'd still like you to explain your suggestion :D